libs/server-sdk/src/client_impl.cpp

```diff
@@ -240,10 +268,53 @@ EvaluationDetail<T> ClientImpl::VariationInternal(Context const& ctx,
         auto error_reason =
             EvaluationReason(EvaluationReason::ErrorKind::kWrongType);
 
+        if (detailed) {
+            event.reason = error_reason;
+        }
+        event_processor_->SendAsync(std::move(event));
         return EvaluationDetail<T>(std::move(default_value), std::nullopt,
                                    error_reason);
     }
 
+    event.value = detail.Value();
+    event.variation = detail.VariationIndex();
+    event.version = flag.Version();
+
+    if (detailed) {
```

Comment:
I think we need another layer here to put the event together. Typically we have an event factory. Then you either make two factories and choose between them based on detail, or you could just have a parameter to a method.

 As is it seems like too much detail at this level.

Also, I think this needs additional experiment handling, but possibly that comes in a later PR.

Reply:
(fixed in future PR)